packages/squiggle-lang/__tests__/TS/KDE.ts

```diff
@@ -0,0 +1,59 @@
+import * as fc from "fast-check";
+import { samplesToContinuousPdf } from "../../src/rescript/Distributions/SampleSetDist/KdeLibrary.js";
+import range from "lodash/range";
+import sumBy from "lodash/sumBy";
+
+// KDE by definition, with triangular kernel max(0, |1-x|)
+const evalKde = (samples: number[], width: number, weight: number, x: number) =>
+  (weight / width) *
+  sumBy(samples, (p) => Math.max(0, 1 - Math.abs((x - p) / width)));
+
+describe("Kernel density estimation", () => {
+  test("should approximately equal naive definition", () => {
+    fc.assert(
+      fc.property(
+        fc.array(fc.double({ min: -10, max: 10, noNaN: true }), {
```

Comment:
Maybe add a comment above to who you chose these values (-10/10, 4, 1e3, 0.1, 3).
"Bounds chosen to make sure there was a medium amount of overlap of kernels."